include/library/spdm_device_secret_lib.h

```diff
@@ -226,3 +226,35 @@ bool libspdm_psk_master_secret_hkdf_expand(
  **/
 bool libspdm_write_certificate_to_nvm(uint8_t slot_id, const void * cert_chain,
                                       size_t cert_chain_size);
+
+/**
+ * save the CSR
+ *
+ * @param[out] csr_len               CSR len for DER format
+ * @param[in]  csr_pointer           csr_pointer is address to store CSR.
+ * @param[in]  file_name             file name
+ *
+ * @retval true                      successfully.
+ * @retval false                     unsuccessfully.
+ **/
+bool libspdm_write_csr_to_nvm(const void * csr_pointer, size_t csr_len, char *file_name);
+
+/**
+ * Gen CSR
+ *
+ * @param[in]      base_hash_algo        hash algo for sign
+ *
+ * @param[out]     csr_len               CSR len for DER format
+ * @param[in]      csr_pointer           For input, csr_pointer is address to store CSR.
+ * @param[out]     csr_pointer           For input, csr_pointer is address for stored CSR.
+ * @param[in]      csr_buffer_size       The size of store CSR buffer.
+ *
+ * @param[in]      requester_info        requester info to gen CSR
+ * @param[in]      requester_info_length The len of requester info
+ *
+ * @retval  true   Success.
+ * @retval  false  Failed to gen CSR.
+ **/
+bool libspdm_gen_csr(uint32_t base_hash_algo, uint32_t base_asym_algo,
+                     size_t *csr_len, uint8_t *csr_pointer, size_t csr_buffer_size,
+                     uint8_t *requester_info, size_t requester_info_length);
```

Comment:
gen_csr() does not mean it can return csr directly.
The flow is:
      if (cached CSR request != NULL) {
           if (current CSR request == cached CSR request) {
                      return (cached signed CSR, success)
           }
      }
      cached CSR request <= current CSR request
     return (NULL, reset_required)